src/components/Header/Header.jsx

```diff
@@ -20,6 +20,7 @@ const Header = ({ role, setSearchInput }) => {
   const signOut = (event) => {
     event.preventDefault();
     localStorage.setItem(process.env.REACT_APP_USER_TOKEN, '');
+    localStorage.setItem("role", '');
```

Comment:
In reference the my comment above, this line is fine, as we're updating the localstorage.